src/mono/wasm/runtime/invoke-js.ts

```diff
@@ -29,6 +29,70 @@ export function mono_wasm_bind_js_function(function_name: MonoStringRef, module_
         const fn = mono_wasm_lookup_function(js_function_name, js_module_name);
         const args_count = get_signature_argument_count(signature);
 
+        const arg_marshalers: Function[] = [];
+        const arg_cleanup: Function[] = [];
+        for (let index = 0; index < args_count; index++) {
+            const sig = get_sig(signature, index + 2);
+            const marshaler_type = get_signature_type(sig);
+            const arg_marshaler = bind_arg_marshal_to_js(sig, marshaler_type, index + 2);
+            arg_marshalers.push(arg_marshaler);
+            arg_cleanup.push(marshaler_type !== MarshalerType.Span
+                ? () => undefined
+                : (js_args: any[]) => {
+                    js_args[index].dispose();
+                });
+        }
+        const res_sig = get_sig(signature, 1);
+        const res_marshaler_type = get_signature_type(res_sig);
+        const res_converter = bind_arg_marshal_to_cs(res_sig, res_marshaler_type, 1);
+
+        const bound_fn = (args: JSMarshalerArguments) => {
+            try {
+                const js_args = [];
```

Comment:
this should definitely be a sized array however, so use `js_args = new Array(args_count)` here and then fill instead of push.
we may also want to reuse a scratch array via try/catch to avoid an alloc on every call, like the old bindings do.